### PR TITLE
Add CI job and build.cmd command for building with FEATURE_STANDALONE_GC

### PR DESCRIPTION
**Pull request by swgillespie**
This build command and CI job build and run tests on the GC built with `FEATURE_STANDALONE_GC`, running weekly to avoid any accidental breakage. I imagine the main use of this job will be to request runs on individual PRs.

### CODE REVIEW

**Comment by swgillespie** [comment]
cc @sergiy-k @adityamandaleeka @Maoni0 ptal?

**Comment by swgillespie** [comment]
@dotnet-bot test CI please

**Comment by swgillespie** [comment]
@dotnet-bot test CI please

**Comment by sergiy-k** [comment]
LGTM

File: netci.groovy [hunk 1]
```diff
@@ -2235,6 +2258,10 @@ combinedScenarios.each { scenario ->
                                     return
                                 }
                                 break
+                            case 'standalone_gc':
+                                if (configuration != 'Release' || configuration != 'Checked') {
```
Comment (adityamandaleeka) [inline netci.groovy]: Should this be &&?

Comment (swgillespie) [inline netci.groovy]: you're right, oops.

**Comment by swgillespie** [comment]
@dotnet-bot test CI please

**Comment by swgillespie** [comment]
@dotnet-bot test CI please

**Comment by swgillespie** [comment]
@dotnet-bot test CI please

**Pull request merged**